pkg/prometheus/server/operator.go

```diff
@@ -951,6 +962,22 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 	return nil
 }
 
+// As the ShardRetentionPolicy feature evolves, should delete will evolve accordingly.
+// For now, shouldDelete just returns the appropriate boolean based on the retention type.
+func (c *Operator) shouldDelete(p *monitoringv1.Prometheus) (bool, error) {
```

Comment:
If I read correctly, the function never returns an error.
```suggestion
func (c *Operator) shouldDelete(p *monitoringv1.Prometheus) bool {
```

Reply:
I think it was added to "future proof" this function. Later on, `shouldDelete` would also involve asking the API server for the object and getting the deletion annotation.
We can leave this too, or remove it if it seems to cause confusion.

Reply:
we can always change the function signature later. I'm more on the side of keeping it consistent (e.g. if it can't return an error, it shouldn't have an error return value).